psydac/linalg/tests/test_block.py

```diff
@@ -1125,6 +1125,69 @@ def test_block_linear_operator_parallel_dot( dtype, n1, n2, p1, p2, P1, P2 ):
     assert np.allclose( Z.blocks[0].toarray(), y1.toarray(), rtol=1e-14, atol=1e-14 )
     assert np.allclose( Z.blocks[1].toarray(), y2.toarray(), rtol=1e-14, atol=1e-14 )
 
+
+# ===============================================================================
+@pytest.mark.parametrize('dtype', [float, complex])
+@pytest.mark.parametrize('n1', [10, 17])
+@pytest.mark.parametrize('n2', [13, 7])
+@pytest.mark.parametrize('p1', [1, 2])
+@pytest.mark.parametrize('p2', [1])
+@pytest.mark.parametrize('s1', [1, 2])
+@pytest.mark.parametrize('s2', [1])
+@pytest.mark.parametrize('P1', [True, False])
+@pytest.mark.parametrize('P2', [True])
+@pytest.mark.parallel
+
+def test_block_vector_2d_parallel_array_to_psydac(dtype, n1, n2, p1, p2, s1, s2, P1, P2):
+    npts = [n1, n2]   
+
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+
+    # Create domain decomposition
+    D = DomainDecomposition(npts, periods=[P1, P2], comm=comm)
+
+    # Partition the points
+    global_starts, global_ends = compute_global_starts_ends(D, npts)
+    C = CartDecomposition(D, npts, global_starts, global_ends, pads=[p1, p2], shifts=[s1, s2])
+
+    # Create Vector spaces and Vectors
+    V = StencilVectorSpace(C, dtype=dtype)
+    W = BlockVectorSpace(V, V)
+    W2 = BlockVectorSpace(W, W, V)
+    x = W.zeros()
+    x2 = W2.zeros()
+
+    # Fill the vector with data
+
+    if dtype == complex:
+        f = lambda i1, i2: 10j * i1 + i2
+    else:
+        f = lambda i1, i2: 10 * i1 + i2
+    for i1 in range(V.starts[0], V.ends[0]+1):
+        for i2 in range(V.starts[1], V.ends[1]+1):
+            x[0][i1, i2] = f(i1, i2)
+            x[1][i1, i2] = f(i1, i2)
```

Comment:
I feel that in the (unlucky) case where the conversion functions mix the components of a block vector, this test will still pass.

I would suggest using a vector with `x[0]` ≠ `x[1]`

Reply:
(the test on `x2` might fail then, but why take the chance?)

Reply:
I agree.

Reply:
Done in [9a93c6d](https://github.com/pyccel/psydac/pull/404/commits/9a93c6d4fe899ee2defce6459e555611fb439bf1)

Reply:
Looks good to me. @campospinto?